### DISCUSSION TITLE
Unable to build from source on Devuan (basically Debian) due to import pyembroidery.exceptions not recognized

### DISCUSSION THREAD

**Discussion by kourama** [inkstitch/inkstitch#3214] [Q&A] [answered]
Hello, I'm wondering if anyone has seen this problem. I'm running python3 in a virtualized environment with venv because Devuan package management prevents pip being used normally. I install the modules needed from requirements.txt and after running make I get:
...
  File "/home/kourama/devel/inkstitch/lib/output.py", line 11, in <module>
    from pyembroidery.exceptions import TooManyColorChangesError
ModuleNotFoundError: No module named 'pyembroidery.exceptions'

and I am unable to fix it. I grabbed the latest pyembroidery source and cannot find an 'exceptions' subpackage, and I can't find the string 'TooManyColorChanges' in any source file within it. I tried checking out ca819c51fc20a694bbbf6a47876dc0546c508509 which is 1 before the relevant change to /lib/output.py, and ran into an error with flask, which I found out through some googling is likely a deprecated requirement because JSON parsing is handled with default libraries now, and anyway I'd prefer to use the latest or at least latest tagged version. I'd appreciate some guidance. thanks.

% uname -a
Linux kourama-pc 6.1.0-21-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.90-1 (2024-05-03) x86_64 GNU/Linux
% python3 --version
Python 3.11.2
% git rev-parse --verify HEAD
9f43fe9eec09c6b01aed196b8c056155b7b44075

**Comment by kaalleen** [comment, marked as answer]
How did you install pyembroidery?
We use the inkstitch fork while the original doesn't include the TooManyColorChangesError.

https://inkstitch.org/developers/inkstitch/manual-setup/

**Reply by kourama** [reply]
Yep. that's it. I cloned & installed the original, but not the inkstitch branch. Thanks!
